planner/core/exhaust_physical_plans.go

```diff
@@ -2021,10 +2021,10 @@ func (p *LogicalJoin) exhaustPhysicalPlans(prop *property.PhysicalProperty) ([]P
 		}
 	})
 
-	if (p.preferJoinType&preferBCJoin) == 0 && (p.preferJoinType&preferShuffleJoin) == 0 && p.preferJoinType > 0 {
-		p.SCtx().GetSessionVars().RaiseWarningWhenMPPEnforced("MPP mode may be blocked because you have used hint to specify a join algorithm which is not supported by mpp now.")
-		if prop.IsFlashProp() {
-			return nil, false, nil
+	if hasMPPJoinHints(p.preferJoinType) {
+		if !checkMPPHintAvailable(p.preferJoinType) {
+			p.SCtx().GetSessionVars().StmtCtx.AppendWarning(ErrInternal.GenWithStack("The MPP join hints are in conflict, and you can only specify join method hints that are currently supported by MPP mode now."))
+			p.preferJoinType = 0
```

Comment:
If the hint is conflict, the principle is all of the conflict hints are invalid. So just invalid the hint and do not return directly.